runtime/compiler/control/CompilationThread.cpp

```diff
@@ -7992,7 +8016,33 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
                }
             }
          }
+#if defined(J9VM_OPT_JITSERVER)
+      else if (eligibleForRemoteAOTNoSCCCompile)
+         {
+         if (TR::Options::getAOTCmdLineOptions()->getOption(TR_ForceAOT))
+            {
+            // TODO: is -Xaot:forceaot recognized when -Xshareclasses:none is specified?
+            canDoRelocatableCompile = true;
+            }
+         else if (!preferLocalComp(entry))
```

Comment:
Would be nice to factor this logic out (although to be fair, this whole method is a bit of a mess and could use some much heavier refactoring...).